### ISSUE TITLE
[Fix] Better error message when the product's slug isn't unique

### ISSUE THREAD

**Issue by jarekmorawski** [woothemes/woocommerce#36317]
**Depends on #36589**

**Related to #36316**

We generate the slug automatically based on the product's name, so when the user creates a product with a name identical to another product and tries to publish it, they'll see a generic error message.

We can improve it by changing the copy of the error message and helping the user fix the link.

**Design**

`5sAIeTRd9Yp7nSCT33BAWz-fi-10611%3A206765&t=reVe66hOXCq7FT7t-4`

<img width="735" alt="image" src="https://user-images.githubusercontent.com/79307566/211017424-191aa659-a362-4d1a-9a91-cc966b692b1a.png">

**Acceptance criteria**

- [ ] On blur of the name field, we will show the product link label and edit link (currently this is not shown until draft is saved)
- [ ] We show a dismissible toast message when the user attempts to publish a product with a duplicated slug.
- [ ] The message stays on the screen until the user interacts with it (dismisses it or clicks the button).
- [ ] The message reads: `Product with this link already exists.`
- [ ] The `Edit link` button opens the `Edit link` modal (the same one that exists below the name input field.
- [ ] [Nice to have] The `Publish` button is disabled until the user edits the link (if the user changes the link and it still isn't unique, we repeat the process).

**Comment by jarekmorawski** [comment]
Related: https://github.com/woocommerce/woocommerce/issues/36774

**Comment by louwie17** [comment]
Closing this, as this was for the old editor, but also has just been addressed in the new editor here: https://github.com/woocommerce/woocommerce/pull/38307
